server/src/test/java/org/elasticsearch/cluster/metadata/MetadataTests.java

```diff
@@ -193,26 +193,6 @@ public void testFindAliasWithExclusionAndOverride() {
         assertThat(aliases.get(1).alias(), equalTo("bb"));
     }
 
-    public void testIndexAndAliasWithSameName() {
```

Comment:
I was trying to find a case where we'd build `IndexMetadata` instances that temporarily breaks this. The one case I could think of was the aliases action where we can add and remove aliases/indices. AFAICS, it does try to do things in order. I think this is OK, but wanted to mention in case you have a case yourself you also want to check.

Obviously we should have tests failling then, but this is pretty old so might not necessarily be covered well.

Reply:
Yea I checked for that and at least couldn't find any broken spots (all of the spots that may throw seem to have safe listeners). If anything this should at least cause a better exception here and there.